autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py

```diff
@@ -337,6 +342,15 @@ def _fit(self, X: Dict[str, Any], y: Any = None, **kwargs: Any) -> torch.nn.Modu
 
             torch.cuda.empty_cache()
 
+        if self.choice.use_swa:
+            # update batch norm statistics
+            swa_utils.update_bn(X['train_data_loader'], self.choice.swa_model)
+            # change model
+            update_model_state_dict_from_swa(X['network'], self.choice.swa_model.state_dict())
```

Comment:
Question: Do you think this implementation is fine? Or should I make an attribute 'swa_network' in base network which will directly predict from the averaged model?

Reply:
okay, then I'll leave it be. Thanks